MISRA.md

```diff
@@ -10,28 +10,28 @@ Deviations from the MISRA standard are listed below:
 | :-: | :-: | :-: |
 | Rule 1.2 | Advisory | Language extenstions are being used in a portable manner for each specific compiler. |
 | Rule 8.7 | Advisory | Functions without an external linkage are being used in other projects. Thus, function declarations are kept. |
-| Rule 19.2 | Advisory | Unions are used to reduce the memory footprint of the stack. |
+| Rule 19.2 | Advisory | Unions are used to reduce the memory footprint and to represent packet formats in the FreeRTOS-Plus-TCP network stack. |
 | Directive 4.6 | Advisory | Non-typedef-ed variables are used in compiler and platform specific manner (portable). |
 | Rule 20.1 | Advisory | #includes are used to include compiler specific attributes to make "[packed structures](https://en.wikipedia.org/wiki/Data_structure_alignment)". |
-| Rule 11.4 | Advisory | Integer to pointer conversion is done at places which are best served in that manner. |
+| Rule 11.4 | Advisory | Integer to pointer conversion is done in various modules to aid in pointer arithmetic and to pass error codes using pointers (such as `FREERTOS_INVALID_SOCKET`). |
 | Rule 11.3 | Required | Pointer conversions are done to access various data members of a given data packet. |
 | Rule 15.4 | Advisory | `break` statements are used in conjucture with `do-while(0)` to reduce `if-else` nesting. |
 | Rule 8.13 | Advisory |  |
-| Rule 10.5 | Advisory | Numerical casting is done only between certain variables and `enums`. It has been verified that the enums used can accomodate variables. |
+| Rule 10.5 | Advisory | Numerical casting is done only for the `ucTCPState` member of `xTCP` structure to cast it to `eIPTCPState_t` enum which has been verified to be big enough to store all values in the `ucTCPState`. |
 | Rule 11.8 | Required | `const` identifier is removed in some cases since certain `struct`s are made to be used with modifiable types. |
 | Rule 14.4 | Required | Condition `0` not having an essentially boolean type is used in portable layer of the Kernel for `taskENTER_CRITICAL` and `taskEXIT_CRITICAL`. |
 | Rule 8.6 | Required | Some functions are declared but not defined since they are to be defined by the user application to provide 'hooks' for certain events. |
 | Rule 2.3 | Advisory |  |
-| Rule 20.10 | Advisory | `#` or `##` is used to create some utility functions for casting variables.  |
+| Rule 20.10 | Advisory | For casting pointers of one type to another, FreeRTOS-Plus-TCP uses utility functions which are created using `##` operator to make signature of functions similar and avoid user the pain of defining functions for various types. |
```

Comment:
Reword this: _For casting between pointer types, FreeRTOS-Plus-TCP uses utility inline functions created using the `##` operator to improve readability._